modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/MeResourceManagerTest.java

```diff
@@ -0,0 +1,1373 @@
+                "      \"op\": \"replace\",\n" + //REPLACE Operation
+                "      \"value\": {\n" +
+                "        \"nickName\": \"shaggy\"\n" +
+                "      }\n" +
+                "    }\n" +
+                "  ],   \n" +
+                " \"name\": {\n" +
+                "    \"givenName\": \"Kim\",\n" +
+                "    \"familyName\": \"Berry\",\n" +
+                "    \"formatted\": \"Kim Berry\"\n" +
+                "  },\n" +
+                "  \"userName\": \"kimjohn  \",\n" +
+                "  \"password\": \"kim123\",\n" +
+                "  \"id\": \"123\",\n" +
+                "  \"emails\": [\n" +
+                "      {\n" +
+                "        \"type\": \"home\",\n" +
+                "        \"value\": \"johnnew@gmail.com\",\n" +
+                "         \"primary\": true\n" +
+                "      }\n" +
+                "  ]\n" +
+                "}";
+    }
+
+    private User getNewUser() throws BadRequestException, CharonException, InternalErrorException {
+
+        String scimObjectString = getNewUserSCIMObjectString();
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+        JSONDecoder decoder = new JSONDecoder();
+
+        return decoder.decodeResource(scimObjectString, schema, new User());
+    }
+
+    @DataProvider(name = "dataForGetSuccess")
+    public Object[][] dataToGetSuccess() throws CharonException, InternalErrorException, BadRequestException {
+
+        User user = getNewUser();
+        String id = user.getId();
+        String name = user.getUserName();
+
+        return new Object[][]{
+                {id, name, "userName", null, 200, user}
+        };
+    }
+
+    @Test(dataProvider = "dataForGetSuccess")
+    public void testGetSuccess(String id, String name, String attributes,
+                               String excludeAttributes, int expectedScimResponseStatus, Object objectUser)
+            throws BadRequestException, NotFoundException, CharonException {
+
+        User user = (User) objectUser;
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+
+        Map<String, Boolean> requiredAttributes = ResourceManagerUtil.getOnlyRequiredAttributesURIs(
+                (SCIMResourceTypeSchema) CopyUtil.deepCopy(schema), attributes, excludeAttributes);
+        mockStatic(AbstractResourceManager.class);
+
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn("https://localhost:9443/scim2/Me");
+        when(AbstractResourceManager.getEncoder()).thenReturn(new JSONEncoder());
+        when(AbstractResourceManager.getDecoder()).thenReturn(new JSONDecoder());
+        when(userManager.getMe(name, requiredAttributes)).thenReturn(user);
+
+        SCIMResponse outputScimResponse = meResourceManager.get(name, userManager, attributes, excludeAttributes);
```

Comment:
better to consider different cases here such as:
- when there are attributes and excludedAttributes params are present in the request.
- when both attributes and excludedAttributes params are not present in the request.
- when only attributes param is present in the request
- when only excludedAttributes param is present in the request

and perform the assertions accordingly

Reply:
Fixed in commit f8e4cf17c52ce61a257e2d77331fff85c69e7f66